Orb slam3 mac

match08 opened the pull request:
Mac support for ORB-SLAM3

match08 committed:
Merge pull request #1 from UZ-SLAMLab/master

fork beta 0.03

match08 committed:
Mac support for ORB-SLAM3

soeroesg commented on include/LoopClosing.h:
```diff
@@ -47,8 +47,12 @@ class LoopClosing
+    //Mac support for ORB-SLAM3
     typedef map<KeyFrame*,g2o::Sim3,std::less<KeyFrame*>,
-        Eigen::aligned_allocator<std::pair<const KeyFrame*, g2o::Sim3> > > KeyFrameAndPose;
+            Eigen::aligned_allocator<std::pair<KeyFrame *const, g2o::Sim3> > > KeyFrameAndPose;
```
@match08 const KeyFrame* is not the same as KeyFrame *const and we need the first one here!

runer112 replied:
The change looks correct to me. I encountered the same issue building on Windows, and this change was the fix for me as, well.

match08 committed:
Mac support for ORB-SLAM3: Viewer and AR Example to MacOS

match08 commented:
> Mac support for ORB-SLAM3

Mac support for ORB-SLAM3 Tested: [Viewer and AR Example to MacOS](https://github.com/match08/ORB_SLAM3/commit/404da9ef5f8829a12b627ec164c8bea8f50ffbb5)

match08 committed:
loadFromBinaryFile added,  load ORB VOC bin fast.

match08 committed:
fixbug: camera index bug

richard-elvira committed:
V0.4: Beta version, 21 April 2021

- Changed OpenCV dynamic matrices to static matrices to speed up the code.

- Capability to measure running time of the system threads.

- Compatibility with OpenCV 4.0 (Requires at least OpenCV 3.0)

- Fixed minor bugs.

# Conflicts:
#	CMakeLists.txt
#	include/LoopClosing.h
#	include/System.h
#	src/LocalMapping.cc
#	src/System.cc
#	src/Viewer.cc

richard-elvira committed:
Beta version, 21 April 2021

- Changed OpenCV dynamic matrices to static matrices to speed up the code.

- Capability to measure running time of the system threads.

- Compatibility with OpenCV 4.0 (Requires at least OpenCV 3.0)

- Fixed minor bugs.

# Conflicts:
#	CMakeLists.txt
#	include/LoopClosing.h
#	include/System.h
#	src/LocalMapping.cc
#	src/System.cc
#	src/Viewer.cc

match08 committed:
Merge branch 'UZ-SLAMLab:master' into master

match08 committed:
update

match08 committed:
Fixed a bug in the relocation function

match08 committed:
Fix Tracking::Relocalization bug.

match08 committed:
fix typo in LocalMapping

match08 committed:
fix typo in LocalMapping

match08 committed:
fixbug: viewer mode

match08 committed:
fixbug: ViewerAR menu_LocalizationMode error

match08 committed:
mono ar update

IamShubhamGupto commented:
@jdtardos 
could this code live on a `macOS` branch? I was able to replicate orb slam3 on my m1 Mac Pro
